### PR TITLE
ref(ui) Update old forms to use new context

### PR DESCRIPTION
**Pull request by markstory**
Update the old form library to not use legacy context anymore. The default value of the new context is odd shaped to maintain backwards compatibility for 'detached' inputs. We have a few forms that use old form input/field components without a wrapping FormContext. This requires the default context value to be undefined so that inputs behave as uncontrolled inputs.

A few examples of these uncontrolled inputs are in the issue saved queries modal, discover query builder, and dashboard widget modal.

### CODE REVIEW

**Review by k-fish — COMMENTED** [review]
Isn't setting `super(props, context)`still legacy context?

**Comment by markstory** [comment]
> Isn't setting super(props, context)still legacy context?

It is. We still use legacy context for router in a few spots :cry:

**Comment by priscilawebdev** [comment]
visual snapshots don't seem right. should we check them?

**Review by priscilawebdev — APPROVED** [review]
lgtm

File: static/app/components/forms/form.tsx [hunk 1]
```diff
@@ -35,22 +35,13 @@ type FormClassState = {
...
   Props extends FormProps = FormProps,
   State extends FormClassState = FormClassState
 > extends React.Component<Props, State> {
-  static childContextTypes = {
-    form: PropTypes.object.isRequired,
-  };
-
   static defaultProps = {
     cancelLabel: t('Cancel'),
     submitLabel: t('Save Changes'),
```
Comment (priscilawebdev) [inline static/app/components/forms/form.tsx:47]: should we update this label to be only "Save"?

**Pull request merged**